com/blogspot/debukkitsblog/Util/Client.java

```diff
@@ -95,77 +89,64 @@ private void startListening(){
-							while(!loginSocket.isConnected()){
-								repairConnection();
-								if(loginSocket.isConnected()){
-									break; //diese, kleinere, innere while-Schleife! -- nicht while(true)
-								}
-								
-								Thread.sleep(5000);
-								repairConnection();
+
+		listeningThread = new Thread(() -> {
+			//Wiederhole ständig die Prozedur:
+			while(true){
+				try{
+					//Bei fehlerhafter Verbindung, diese reparieren
+					if(loginSocket != null && !loginSocket.isConnected()){
+						while(!loginSocket.isConnected()){
+							repairConnection();
+							if(loginSocket.isConnected()){
+								break; //diese, kleinere, innere while-Schleife! -- nicht while(true)
 							}
+
+							Thread.sleep(5000);
+							repairConnection();
 						}
-						
-						onConnectionGood();
-						
-						//Auf eingehende Nachricht warten und diese bei Eintreffen lesen
-						ObjectInputStream ois = new ObjectInputStream(loginSocket.getInputStream());
-						Object raw = ois.readObject();
-						
-						//Nachricht auswerten
-						if(raw instanceof Datapackage){
-							final Datapackage msg = (Datapackage) raw;
-							
-							for(final String current : idMethods.keySet()){
-								if(msg.id().equalsIgnoreCase(current)){
-									System.out.println("[Client] Message received. Executing method for '" + msg.id() + "'...");
-									new Thread(new Runnable(){
-										public void run(){
-											idMethods.get(current).run(msg, loginSocket);
-										}
-									}).start();
-									break;
-								}
+					}
+
+					onConnectionGood();
+
+					//Auf eingehende Nachricht warten und diese bei Eintreffen lesen
+					ObjectInputStream ois = new ObjectInputStream(loginSocket.getInputStream());
+					Object raw = ois.readObject();
+
+					//Nachricht auswerten
+					if(raw instanceof Datapackage) {
+						final Datapackage msg = (Datapackage) raw;
+
+						for (final String current : idMethods.keySet()) {
+							if (msg.id().equalsIgnoreCase(current)) {
+								System.out.println("[Client] Message received. Executing method for '" + msg.id() + "'...");
+								new Thread(() -> {idMethods.get(current).run(msg, loginSocket);}).start();
+								break;
 							}
-															
 						}
-						
-					} catch(Exception ex){					
-						if(ex.getMessage().equals("Connection reset")){
-							onConnectionProblem();
-							System.err.println("Server offline.");							
-							if((++errorCount > 30) && autoKill){
-								System.err.println("Server dauerhaft nicht erreichbar, beende.");
-								System.exit(0);
-							} else {
-								repairConnection();
-							}							
+					}
+				} catch(Exception ex){
+					if(ex.getMessage().equals("Connection reset")){
+						onConnectionProblem();
+						System.err.println("[WARNING] Server offline.");
+						if((++errorCount > 30) && autoKill){
+							System.err.println("[ERROR] Server dauerhaft nicht erreichbar, beende.");
+							System.exit(0);
 						} else {
-							ex.printStackTrace();
```

Comment:
I would prefer, if not on every disconnection (eg. due to poor mobile network connection) the entire stack was printed, but only an information to the user that the connection is lost. The application is to eventually work for the console, too, and do not clutter it every 5 seconds, only because the connection breaks off for a very short period of time.